pkg/scheduler/factory/factory.go

```diff
@@ -992,7 +992,10 @@ func (c *configFactory) updateNodeInCache(oldObj, newObj interface{}) {
 	}
 
 	c.invalidateCachedPredicatesOnNodeUpdate(newNode, oldNode)
-	c.podQueue.MoveAllToActiveQueue()
+	// Only activate unschedulable pods if the node became more schedulable.
+	if c.podQueue.HasUnschedulablePods() && nodeSchedulingPropertiesChanged(newNode, oldNode) {
```

Comment:
You should instead do the following:

```suggestion
	if !c.podQueue.HasUnschedulablePods() || nodeSchedulingPropertiesChanged(newNode, oldNode) {
	    c.podQueue.MoveAllToActiveQueue()
	}
```

It may seem odd to move pods when there is nothing in the unschedulable queue, but we should do so, because there may be a pod that the scheduler is currently processing and it may be determined "unschedulable". The scheduling queue has a mechanism to retry such pod when a "move to active queue" event happens. So, what the above optimization does is that when there is no unschedulable pod in the unschedulable queue, it skips the node comparison and send a move to active queue request.